Removes bestF from Master Controller Crash Report Filtering (and uses voodoo magic to help cleaning up down the line)

san7890 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

We removed bestF from datums back in #2701, but this still remained in the filter. Bit silly, let's get rid of this old cobweb.

I couldn't find anything suggesting that the master_controller could get a statclick variable, but that's actually a term we use in the modern day so I'm not going to bug it since it's benign. The other five are still BYOND-level variables.

Addendum: We now also gather the variables we don't want to show up in a new proc that calls `NAMEOF` to ensure that the vars we search for are still valid and not cobwebs, so we compile error should one of these variables disappear (very future proofy, I like it). Report and rebooting the MC goes off effortlessly in the new system by my local testing, so we should be alright.

<!-- Describe The Pull Request. Please be sure every change is documented or this can delay review and even discourage maintainers from merging your PR! -->

## Why It's Good For The Game

We fixed this stupid datum-level definition eight years ago but never cleaned it out of every crevice, very sad. Let's ensure this doesn't happen again.

<!-- Argue for the merits of your changes and how they benefit the game, especially if they are controversial and/or far reaching. If you can't actually explain WHY what you are doing will improve the game, then it probably isn't good for the game in the first place. -->

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

Nothing particularly concerning to players.

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

san7890 committed:
Removes bestF from Master Controller Crash Filtering

We removed bestF from datums back in #2701, but this still remained in the filter. Bit silly, let's get rid of this old cobweb.

I couldn't find anything suggesting that the master_controller could get a statclick variable, but that's actually a term we use in the modern day so I'm not going to bug it since it's benign. The other five are still BYOND-level variables.

MrStonedOne commented:
use `NAMEOF` (https://github.com/tgstation/tgstation/pull/70625) and then this will compile error next time.

note: nameof likely does not work within switch statements, you'll likely have to move to a find in list.

san7890 committed:
imbues deep voodoo magicks

this is really swell actually

san7890 commented:
```txt
[2022-10-19 00:47:27.196] ## DEBUG: Tue Oct 18 18:47:14 2022 MC restarted. Reports:
 - 	 datum_flags = 0
 - 	 datum_components = 
 - 	 comp_lookup = /list
 - 	 tgui_shared_states = 
 - 	 processing = 1
 - 	 map_loading = 0
 - 	 random_seed = 5.10534e+008
 - 	 iteration = 482
 - 	 stack_end_detector = /datum/stack_end_detector(/datum/stack_end_detector)
 - 	 last_run = 402.5
 - 	 subsystems = /list
 - 	 init_stage_completed = 2
 - 	 init_timeofday = 28015
 - 	 init_time = 161
 - 	 tickdrift = 116.42
 - 	 sleep_delta = 1
 - 	 skip_ticks = 0
 - 	 make_runtime = 0
 - 	 initializations_finished_with_no_players_logged_in = 0
 - 	 last_type_processed = Asset Loading(/datum/controller/subsystem/asset_loading)
 - 	 queue_head = Asset Loading(/datum/controller/subsystem/asset_loading)
 - 	 queue_tail = Wardrobe(/datum/controller/subsystem/wardrobe)
 - 	 queue_priority_count = 105
 - 	 queue_priority_count_bg = 50
 - 	 current_runlevel = 3
 - 	 sleep_offline_after_initializations = 1
 - 	 current_initializing_subsystem = 
 - 	 restart_clear = 0
 - 	 restart_timeout = 0
 - 	 restart_count = 0
 - 	 gc_destroyed = 
 - 	 active_timers = 
 - 	 status_traits = 
 - 	 signal_procs = 
 - 	 weak_reference = 
 - 	 cooldowns = 
 - 	 __auxtools_weakref_id = 
 - 	 current_ticklimit = 97.8761
```
Got this report by calling `recover()` on the MC on the new system, seems to be filtering a-okay.

san7890 committed:
GODDAMMIT I ACCIDENTALLY COMMITED THE TEST CASE

at least ur 100% certain i'm not making this up now

MrStonedOne commented on code/controllers/master.dm:
```diff
@@ -161,17 +161,18 @@ GLOBAL_REAL(Master, /datum/controller/master) = new
-		switch (varname)
-			if("name", "tag", "bestF", "type", "parent_type", "vars", "statclick") // Built-in junk.
-				continue
+	var/list/master_attributes = Master.vars
+	var/list/filtered_variables = gather_filtered_variables()
+	for (var/varname in master_attributes)
+		if(filtered_variables.Find(varname)) // Built-in junk.
+			continue
```
```suggestion
	var/list/master_attributes = Master.vars
	var/list/filtered_variables = list(NAMEOF(src, name), NAMEOF(src, tag), NAMEOF(src, type), NAMEOF(src, parent_type) NAMEOF(src, vars), NAMEOF(src, statclick))
	for (var/varname in master_attributes - filtered_variables)
```

NAMEOF returns the string version of the varname.

```dm
world.log << json_encode(NAMEOF(src, statclick)))
```

returns `"statclick"` ie, NAMEOF even adds the "" for you.

san7890 committed:
Update code/controllers/master.dm

i'm going to span out the list in a second

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

san7890 committed:
patches and cruft handling

deletes the proc i made, spans out MSO's list

accidentally deleted an N

and alphabetize

san7890 commented:
Cleaned up some cruft, reporting still works as intended

<details>
<summary>see spoiler for the report</summary>

```txt
[2022-10-19 06:06:49.175] ## DEBUG: Wed Oct 19 00:06:41 2022 MC restarted. Reports:
 -      datum_flags = 0
 -      datum_components = 
 -      comp_lookup = 
 -      tgui_shared_states = 
 -      processing = 1
 -      map_loading = 0
 -      random_seed = 7.62063e+008
 -      iteration = 464
 -      stack_end_detector = /datum/stack_end_detector(/datum/stack_end_detector)
 -      last_run = 391
 -      subsystems = /list
 -      init_stage_completed = 2
 -      init_timeofday = 219718
 -      init_time = 158.5
 -      tickdrift = 84.9692
 -      sleep_delta = 1
 -      skip_ticks = 0
 -      make_runtime = 0
 -      initializations_finished_with_no_players_logged_in = 0
 -      last_type_processed = Atmospherics(/datum/controller/subsystem/air)
 -      queue_head = Asset Loading(/datum/controller/subsystem/asset_loading)
 -      queue_tail = Wardrobe(/datum/controller/subsystem/wardrobe)
 -      queue_priority_count = 105
 -      queue_priority_count_bg = 50
 -      current_runlevel = 3
 -      sleep_offline_after_initializations = 1
 -      current_initializing_subsystem = 
 -      restart_clear = 0
 -      restart_timeout = 0
 -      restart_count = 0
 -      gc_destroyed = 
 -      active_timers = 
 -      status_traits = 
 -      signal_procs = 
 -      weak_reference = 
 -      cooldowns = 
 -      __auxtools_weakref_id = 
 -      current_ticklimit = 97.9528
 ```
</details>
